packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -338,6 +320,27 @@ class MDnsVmServiceDiscovery {
     }
   }
 
+  String _getAuthCode(String txtRecord) {
+    const String authCodePrefix = 'authCode=';
+    String? raw;
+    for (final String record in txtRecord.split('\n')) {
+      if (record.startsWith(authCodePrefix)) {
+        raw = record;
+        break;
+      }
+    }
```

Comment:
I know you just moved this part, but can you make this use `LineSplitter` instead? `import 'convert.dart'` (not `dart:convert`). You can also use `where` (I wish `whereFirst` worked well with nullable values, but it doesn't...), up to you.
```suggestion
    final Iterable<String> matchingRecords =
        LineSplitter.split(txt.first.text).where((String record) => record.startsWith(authCodePrefix));
    for (final String record in matchingRecords) {
      raw = record;
      break;
    }
```